lib/utils.py

```diff
@@ -25,31 +25,62 @@ def build(command: List[str], directory: str):
         return False
     return True
 
+def build_architecture(arch, cross, num_cpus, target):
+    print(f"Building {arch}")
+    data = ["make", f"ARCH={arch}", cross, "-j", str(num_cpus), "defconfig", target]
+    subprocess.check_output(data)
+    print(f"{arch} works")
+
 def build_check(target: Path) -> bool:
     '''Checks if the linux kernel builds properly'''
 
+    specific_arch = None
+    path_parts = target.parts
+    if 'arch' in path_parts:
+        arch_index = path_parts.index('arch')
+        specific_arch = path_parts[arch_index + 1]
+
     num_cpus = len(os.sched_getaffinity(0))
     shutil.copy('.config', '.tmp.config')
     try:
-        print("Building arm")
-        data = ["make", "ARCH=arm", "LLVM=1", "-j", str(num_cpus), "defconfig", target]
-        subprocess.check_output(data)
-        print("arm works")
-
-        print("Building arm64")
-        data[1] = "ARCH=arm64"
-        subprocess.check_output(data)
-        print("arm64 works")
-
-        print("Building riscv")
-        data[1] = "ARCH=riscv"
-        subprocess.check_output(data)
-        print("riscv works")
-
-        print("Building x86")
-        data[1] = "ARCH=x86"
-        subprocess.check_output(data)
-        print("x86 works")
+        # TODO: S390 has Clang support but for Clang-15, Debian currently supports Clang-14
+        clang_archs = ["arm", "arm64", "mips", "riscv", "powerpc", "x86", "um"]
+        for arch in clang_archs:
+            if specific_arch and specific_arch != arch:
+                continue
+            build_architecture(arch, 'LLVM=1', num_cpus, target)
+
+        # Architecture crosstools found here:  https://mirrors.edge.kernel.org/pub/tools/crosstool
+        architectures = ['alpha', 'arc', 'csky', 'hppa', 'hppa64', 'i386', 'loongarch64', 'm68k', 'microblaze',
+                         'mips64', 'nios2', 'or1k', 's390', 'sh2', 'sh4', 'sparc', 'sparc64', 'xtensa']
+
+        name_mappings = {
+            'alpha': 'alpha',           # DEC Alpha
+            'arc': 'arc',               # Argonaut RISC Core
+            'csky': 'csky',             # C-SKY
+            'hppa': 'parisc',           # HP Precision Architecture
+            'hppa64': 'parisc',         # HP Precision Architecture (64-bit)
+            'i386': 'x86',              # Intel 386 (often grouped under 'x86')
```

Comment:
You should be able to build 32b x86 with llvm; `make LLVM=1 ARCH=i386 defconfig all`. Move this arch to the llvm block above?